File: gameserver/src/main/java/l2s/gameserver/skills/effects/i_set_skill.java
package l2s.gameserver.skills.effects;

import l2s.gameserver.data.xml.holder.SkillHolder;
import l2s.gameserver.model.Creature;
import l2s.gameserver.model.Player;
import l2s.gameserver.model.Skill;
import l2s.gameserver.skills.SkillEntry;
import l2s.gameserver.templates.skill.EffectTemplate;

public final class i_set_skill extends i_abstract_effect
{
	private final SkillEntry _skill;

	public i_set_skill(Creature creature, Creature target, Skill skill, boolean reflected, EffectTemplate template)
	{
		super(creature, target, skill, reflected, template);
		int[] temp = template.getParam().getIntegerArray("skill", "-");
		_skill = SkillHolder.getInstance().getSkillEntry(temp[0], temp.length >= 2 ? temp[1] : 1);
	}

	@Override
	public boolean checkCondition()
	{
		return _skill != null && super.checkCondition();
	}

	@Override
	public void instantUse()
	{
		Player player = getEffected().getPlayer();
		player.addSkill(_skill, true);
		player.updateStats();
		player.sendSkillList();
		player.updateSkillShortcuts(_skill.getId(), _skill.getLevel());
	}
}

File: gameserver/src/main/java/l2s/gameserver/handler/effects/impl/instant/retail/i_set_skill.kt
package l2s.gameserver.handler.effects.impl.instant.retail

import l2s.gameserver.handler.effects.impl.instant.i_abstract_effect
import l2s.gameserver.model.Creature
import l2s.gameserver.model.actor.instances.player.Cubic
import l2s.gameserver.skills.SkillEntry
import l2s.gameserver.skills.SkillEntryType
import l2s.gameserver.templates.skill.EffectTemplate
import java.util.concurrent.atomic.AtomicBoolean

/**
 * Set Skill effect implementation.
 * @author Zoey76
 */
class i_set_skill(template: EffectTemplate) : i_abstract_effect(template) {

    private val skill: SkillEntry

    init {
        val param = params.getIntegerArray("skill", "-")
        val level = if (param.size >= 2) param[1] else 1
        val temp = SkillEntry.makeSkillEntry(SkillEntryType.NONE, param[0], level)
        skill = requireNotNull(temp)
    }

    override fun instantUse(caster: Creature, target: Creature, soulShotUsed: AtomicBoolean, reflected: Boolean, cubic: Cubic) {
        val player = target.player ?: return

        player.addSkill(skill, true)
        player.sendSkillList()
        player.updateSkillShortcuts(skill.id, skill.level)
    }
}